miio/walkingpad.py

```diff
@@ -0,0 +1,189 @@
+    # raw_command get_prop '["dist"]' returns [1869]
+    # raw_command get_prop '["cal"]' returns [67340]
+    # raw_command get_prop '["goal"]' returns [0, 60]
+    # raw_command get_prop '["max"]' returns [6.0]
+    # raw_command get_prop '["initial"]' returns [3]
+    # raw_command get_prop '["offline"]' returns [0]
+    # raw_command get_prop '["sensitivity"]' returns [2]
+    # raw_command get_prop '["sp"]' returns [1.0]
+    # raw_command get_prop '["start_speed"]' returns [1.0]
+    # raw_command get_prop '["auto"]' returns [1]
+    # raw_command get_prop '["disp"]' returns [19]
+    # raw_command get_prop '["lock"]' returns [0]
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+
+        # NOTE: Only 1 property can be requested at the same time
+        self.data = data
+
+    @property
+    def power(self) -> str:
+        """Power state."""
+        return self.data["power"]
+
+    @property
+    def is_on(self) -> bool:
+        """True if the device is turned on."""
+        return self.power == "on"
+
+    @property
+    def time(self) -> int:
+        """Current lock status."""
+        return self.data["time"]
+
+    @property
+    def speed(self) -> float:
+        """Current speed."""
+        return self.data["sp"]
+
+    @property
+    def mode(self) -> int:
+        """Current mode."""
+        return self.data["mode"]
+
+    @property
+    def step(self) -> int:
+        """Current steps."""
+        return self.data["step"]
+
+    @property
+    def distance(self) -> int:
+        """Current distance."""
+        return self.data["dist"]
+
+    @property
+    def calories(self) -> int:
+        """Current distance."""
+        return self.data["cal"]
+
+
+class Walkingpad(Device):
+    """Main class representing Xiaomi Walkingpad."""
+
+    # TODO: - Auto On/Off Not Supported
+    #       - Adjust Scenes with Wall Switch Not Supported
+
+    @command(
+        default_output=format_output(
+            "",
+            "Mode: {result.mode}\n"
+            "Time: {result.time}\n"
+            "Steps: {result.step}\n"
+            "Speed: {result.speed}\n"
+            "Distance: {result.distance}\n"
+            "Calories: {result.calories}  ",
+        )
+    )
+    def status(self) -> WalkingpadStatus:
+        """Retrieve properties."""
+
+        properties_received = []
+        values_received = []
+
+        # Walkingpad A1 allows you to retrieve a subset of values with "all"
+        # eg ['mode:1', 'time:1387', 'sp:3.0', 'dist:1150', 'cal:71710', 'step:2117']
+
+        properties = ["all"]
+
+        # Walkingpad A1 only allows 1 property to be read at a time
+
+        values = self.get_properties(properties, max_properties=1)
+
+        # When running the tests, for some reason the list provided is passed within another list, so I take
+        # care of this here.
+
+        if any(isinstance(el, list) for el in values):
+            values = values[0]
```

Comment:
You could do a check like this to make sure that a non-empty list is returned:
```
if not values:
    raise ...
```

and  then simply do `values = values.pop()` afterwards.

Reply:
This one I am not sure I fully understood your proposal, but I have worked to simplify / improve:

        if len(values) <= 1:
            values = values.pop()

Reply:
What I meant with that is you can use an if clause to see if the variable is not None or an empty list, in which case you can simply do the pop to access its first element.

You mention in the comment, that it happens during the test. Do you mean during unit tests or in the data reported by the device?

Reply:
OK, thanks for the tips. In the end I was solving the wrong problem. I needed to create different behaviour for the dummy device to replicate how the physical device behaves when queried with the 'all' parameter. 

I did some research and figured out that I needed to override the dummy class in the test, which I have done to deal with this case. Now the code doesn't contain a weird workaround for the test, which I guess is much better :)